source/elements/oneMKL/source/domains/dft/config_params/data_layouts.rst

```diff
@@ -0,0 +1,191 @@
+        * -   ``COMPLEX`` template :ref:`onemkl_dft_enum_domain` and :ref:`onemkl_dft_complex_storage_complex_complex`
+          -   Complex
+          -   Complex
+        * -   ``COMPLEX`` template :ref:`onemkl_dft_enum_domain` and :ref:`onemkl_dft_complex_storage_real_real`
+          -   Real
+          -   Real
+        * -   ``REAL`` template :ref:`onemkl_dft_enum_domain`, :ref:`onemkl_dft_real_storage_real_real` and :ref:`onemkl_dft_conjugate_even_storage_complex_complex`
+          -   Real
+          -   Complex
+
+
+.. _onemkl_dft_fwd_bwd_strides:
+
+.. rubric:: FWD_STRIDE and BWD_STRIDES
+
+The values :math:`s_0, s_1, \dots, s_d` in :eq:`eq_idx_data_layout` here above
+are communicated as elements, in that order, of a :math:`(d+1)`-long
+``std::vector<std::int64_t>`` value for the configuration parameter
+``config_param::FWD_STRIDES`` (resp. ``config_param::BWD_STRIDES``). The element
+:math:`s_0` represents an absolute offset (or "displacement") in the data set(s)
+while the subsequent element(s) :math:`s_j\ (j > 0)` are generalized strides.
+
+The default values set for the forward and backward strides correspond to the
+data layout configurations for unbatched, in-place transforms using unit stride
+along the leading dimension with no offset (and minimal padding in forward
+domain in case of descriptors of ``REAL`` template :ref:`onemkl_dft_enum_domain`,
+aligning with the :ref:`requirements for in-place
+transforms<onemkl_dft_data_layout_requirements>`). In other words, the default
+values are :math:`s_0 = 0`, :math:`s_d = 1` and, for multi-dimensional
+transforms, *i.e.*, if :math:`d > 1`,
+
+- :math:`s_{d-1} = n_{d}` for descriptors of ``COMPLEX`` template :ref:`onemkl_dft_enum_domain`;
+
+- :math:`s_{d-1} = \alpha \left(\lfloor \frac{n_{d}}{2} \rfloor + 1\right)`, wherein :math:`\alpha = 2` (resp. :math:`\alpha = 1`) for forward (resp. backward) strides, of descriptors of ``REAL`` template :ref:`onemkl_dft_enum_domain`;
+
+and :math:`s_k = n_{k+1} * s_{k+1}` for :math:`1 \leq k < d - 1` if :math:`d > 2`.
+
+.. _onemkl_dft_fwd_bwd_distance:
+
+.. rubric:: FWD_DISTANCE and BWD_DISTANCE
+
+The value :math:`l` in :eq:`eq_idx_data_layout` here above is communicated as an
+``std::int64_t`` value for the configuration parameter
+``config_param::FWD_DISTANCE`` (resp. ``config_param::BWD_DISTANCE``). This
+configuration parameter is irrelevant for unbatched transforms, *i.e.*, for
+descriptors set to handle a
+:ref:`number of transforms<onemkl_dft_config_number_of_transforms>` equal to
+:math:`1` (default value). The configuration parameters
+``config_param::FWD_DISTANCE`` and ``config_param::BWD_DISTANCE`` must be set
+for descriptors configured to compute more than one transform, *i.e.*, if the
+:ref:`number of transforms<onemkl_dft_config_number_of_transforms>` is larger
+than :math:`1`. The default values for ``config_param::FWD_DISTANCE`` and
+``config_param::BWD_DISTANCE`` are :math:`0` regardless of the descriptor's
+nature.
```

Comment:
Is the change of the default value for distances from 1 to 0 intentional?

Reply:
Yes it is. The reasoning is to **explicitly require their values to be set** by users in case of batched transforms by setting the default distances to the only values that can't possibly be "consistent" for any batched DFT regardless of the stride values, *i.e.*, $0$. In the form of the specs as updated herein, this is now explicitly specified in the updated table of the "config_param" section of the page entitled "DFT-related scoped enumeration types" and further developed in section "Configuring data layouts for batched transforms" of the page entitled "Configuration of Data Layouts".

The main motivation is that, in my understanding, [rocFFT](https://rocm.docs.amd.com/projects/hipfort/en/docs-5.4.1/doxygen/html/interfacehipfort__rocfft_1_1rocfft__plan__description__set__data__layout.html), [cuFFT](https://docs.nvidia.com/cuda/cufft/index.html#advanced-data-layout), [FFTW](https://www.fftw.org/fftw3_doc/Advanced-Complex-DFTs.html) and even closed-source oneMKL require distances to be explicitly set by users. NB: for closed-source oneMKL, the default value is actually $0$ therefore _indirectly_ requesting users to set the distances in case of batched DFTs (error at commit to expect otherwise). I truly can't explain how this value of $1$ ever made it in here and why...
A secondary motivation is that changing closed-source oneMKL's default behavior for no other reason than aligning it with that value of $1$ (if unchanged) would result in truly nightmare-ish changes as such a default value is a convenient assumption in a lot of algorithm selection checks as they are implemented: they happen to be valid as is too for unbatched transforms if the distances are $0$, _i.e._, if left as default, but would no longer hold if that were to change (requiring seemingly insignificant changes in many if not all algorithm selection checks to prevent performance regressions).

Finally, if there exists reluctance to change this default value to $0$, I don't think that leaving it as $1$ makes sense either: now that we're introducing fwd and bwd strides, we _might_ want to consider a default value that would be appropriate for default strides and in-place operations. But we'd end up in an arguably rather inconvenient state where users would need only ``NUMBER_OF_TRANSFORMS`` to be set for batched real in-place while they'd need to reset ``FWD_DISTANCE``, too, for out-of-place real transforms of standard layout. In my opinion, we might as well make sure that users know that they're expected to set all ``NUMBER_OF_TRANSFORMS`` and ``(F|B)WD_DISTANCE``

Reply:
Just adding my on upvote for this change. Abstractly I think zero's a better choice for both the implementers and the users since it makes the spec requirement that users set the distances algorithmically enforceable. As mentioned, most of the backends already have that requirement.

Reply:
I was just checking if this is intentional or not. I think arguments for this change are good. 

Although, I would prefer to have this be the value that makes sense for default strides, assuming no gaps in the data.

Reply:
This could be a sensible choice, too, indeed, except for real in-place and `config_value::INPLACE` is the default value for `config_param::PLACEMENT`. My opinion is that this could deserve a dedicated discussion beyond this PR, I'm suggesting to resolve as is for now, due to time constraints.